events/handler.go

```diff
@@ -273,7 +297,18 @@ func (self *events) AddEvent(e *Event) error {
 	self.updateEventList(e)
 	watchesToSend := self.findValidWatchers(e)
 	for _, watchObject := range watchesToSend {
-		watchObject.channel <- e
+		watchObject.eventChannel.GetChannel() <- e
+	}
+	return nil
+}
+
+func (self *events) RemoveDeadChannels() error {
```

Comment:
I'd rather us not have to have this is we can. To remove the channels in a timely manner we may need to have the event manager spin up a thread that listens to all stops and does the removal.

So we'd start a cleanup thread that listens on a cleanup channel, when someone calls Stop() it sends a request to the cleanup thread to remove that channel. The thread wakes up and removed it.